.maintain/ensure-deps.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+# The script is meant to check if the rules regarding packages
+# dependencies are satisfied.
+# The general format is:
+# [top-lvl-dir] MESSAGE/[other-top-dir]
+
+# For instance no crate within `./client` directory
+# is allowed to import any crate with a directory path containing `frame`.
+# Such rule is just: `client crates must not depend on anything in /frame`.
+
+# The script should be run from the main repo directory!
+
+set -u
+
+# HARD FAILING
+MUST_NOT=(
+	"client crates must not depend on anything in /frame"
+	"client crates must not depend on anything in /node"
+	"frame crates must not depend on anything in /node"
+	"frame crates must not depend on anything in /client"
+	"primitives crates must not depend on anything in /frame"
+)
+
+# ONLY DISPLAYED, script still succeeds
+PLEASE_DONT=(
+	"primitives crates should not depend on anything in /client"
+)
+
+VIOLATIONS=()
+PACKAGES=()
+
+function check_rule() {
+	rule=$1
+	from=$(echo $rule | cut -f1 -d\ )
+	to=$(echo $rule | cut -f2 -d\/)
+
+	cd $from
+	echo "Checking rule '$rule'"
+	packages=$(find -name Cargo.toml | xargs grep -wn "path.*\.\.\/$to")
```

Comment:
same here, the path..

We need some customize to this script

Reply:
Under the current rule, this will be fine.